packages/arbitrum-retryables/src/mapping.ts

```diff
@@ -0,0 +1,164 @@
+
+/**
+ * Extend retryable's timeout
+ * @param event
+ */
+export function handleLifetimeExtended(event: LifetimeExtended): void {
+  let ticketId = event.params.ticketId;
+  let entity = Retryable.load(ticketId.toHexString());
+  if (!entity) {
+    log.critical("Missed a retryable ticket somewhere!", []);
+    throw new Error("No retryable ticket");
+  }
+  entity.timeoutTimestamp = entity.timeoutTimestamp.plus(event.params.newTimeout);
+  entity.save();
+}
+
+/**
+ * Deduce if redeem was successful by doing contract call - if redeem was successful call will fail
+ * because ticket has been deleted from queue.
+ * @param event
+ */
+export function handleRedeemScheduled(event: RedeemScheduled): void {
+  let ticketId = event.params.ticketId;
+  let entity = Retryable.load(ticketId.toHexString());
+  if (!entity) {
+    log.critical("Missed a retryable ticket somewhere!", []);
+    throw new Error("No retryable ticket");
+  }
+
+  const stats = getOrCreateTotalRetryableStats();
+
+  const prevStatus = entity.status;
+  const redeemSuccessful = isRedeemSuccessful(ArbRetryableTxContract.bind(event.address), ticketId);
+  if (redeemSuccessful) {
+    if (prevStatus == "Created") {
+      entity.isAutoRedeemed = true;
+      stats.autoRedeemed = stats.autoRedeemed.plus(BigInt.fromI32(1));
+    }
+    if (prevStatus == "RedeemFailed") {
+      stats.failedToRedeem = stats.failedToRedeem.minus(BigInt.fromI32(1));
+    }
+    entity.status = "Redeemed";
+    entity.redeemedAtTimestamp = event.block.timestamp;
+    stats.successfullyRedeemed = stats.successfullyRedeemed.plus(BigInt.fromI32(1));
+  } else {
+    if (prevStatus != "RedeemFailed") {
+      stats.failedToRedeem = stats.failedToRedeem.plus(BigInt.fromI32(1));
+    }
+    entity.isAutoRedeemed = false;
+    entity.status = "RedeemFailed";
+  }
+
+  entity.retryTxHash = event.params.retryTxHash;
+  entity.sequenceNum = event.params.sequenceNum;
+  entity.donatedGas = event.params.donatedGas;
+  entity.gasDonor = event.params.gasDonor;
+  entity.maxRefund = event.params.maxRefund;
+  entity.submissionFeeRefund = event.params.submissionFeeRefund;
+  entity.save();
+  stats.save();
+
+  // decode and save retyrable submission param
+  decodeRetryableParamsFromTxInput(entity, event.transaction.input);
+}
+
+function isRedeemSuccessful(contract: ArbRetryableTxContract, ticketId: Bytes): boolean {
+  const beneficiaryCall = contract.try_getBeneficiary(ticketId);
+  return beneficiaryCall.reverted;
+}
+
+function getOrCreateTotalRetryableStats(): TotalRetryableStats {
+  let stats = TotalRetryableStats.load("NitroStats");
+  if (stats != null) {
+    return stats as TotalRetryableStats;
+  }
+
+  stats = new TotalRetryableStats("NitroStats");
+  stats.totalCreated = BigInt.fromI32(0);
+  stats.autoRedeemed = BigInt.fromI32(0);
+  stats.successfullyRedeemed = BigInt.fromI32(0);
+  stats.failedToRedeem = BigInt.fromI32(0);
+  stats.canceled = BigInt.fromI32(0);
+  stats.save();
+
+  return stats;
+}
```

Comment:
These may be off if we decided not to sync from genesis.

Reply:
True, atm we're syncing from the Nitro genesis block so should be ok. I'll mention that in docs/README